hw05_parallel_execution/run.go

```diff
@@ -2,14 +2,71 @@ package hw05parallelexecution
 
 import (
 	"errors"
+	"sync"
 )
 
 var ErrErrorsLimitExceeded = errors.New("errors limit exceeded")
 
 type Task func() error
 
+type errorCount struct {
+	mu          *sync.Mutex
+	errorNumber int
+	err         error
+}
+
 // Run starts tasks in n goroutines and stops its work when receiving m errors from tasks.
+
 func Run(tasks []Task, n, m int) error {
-	// Place your code here.
-	return nil
+	taskChan := make(chan Task, len(tasks))
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	go addTasks(&taskChan, tasks, &wg)
```

Comment:
```suggestion
	addTasks(&taskChan, tasks, &wg)
```
горутина нам тут не нужна